care/emr/resources/medication/administration/spec.py

```diff
@@ -0,0 +1,198 @@
+        ...,
+        description="Represents the current status of the medication administration",
+    )
+    status_reason: Coding | None = Field(
+        None,
+        description="The reason why the medication was not administered",
+        json_schema_extra={"slug": CARE_MEDICATION_VALUESET.slug},
+    )
+    category: MedicationAdministrationCategory | None = Field(
+        None,
+        description="The category of the medication administration",
+    )
+
+    medication: Coding = Field(
+        ...,
+        description="The medication that was taken",
+        json_schema_extra={"slug": CARE_MEDICATION_VALUESET.slug},
+    )
+
+    authored_on: datetime | None = Field(
+        None,
+        description="When request was initially authored",
+    )
+    occurrence_period_start: datetime = Field(
+        description="When the medication was administration started",
+    )
+    occurrence_period_end: datetime | None = Field(
+        None,
+        description="When the medication administration ended. If not provided, it is assumed to be ongoing",
+    )
+    recorded: datetime | None = Field(
+        None,
+        description="When administration was recorded",
+    )
+
+    encounter: UUID4 = Field(
+        ...,
+        description="The encounter where the administration was noted",
+    )
+    request: UUID4 = Field(
+        ...,
+        description="The medication request under which the administration was made",
+    )
+
+    performer: list[MedicationAdministrationPerformer] | None = Field(
+        None,
+        description="Who administered the medication",
+    )
+    dosage: Dosage | None = Field(
+        None,
+        description="The dosage of the medication",
+    )
+
+    note: str | None = Field(
+        None,
+        description="Any additional notes about the medication",
+    )
+
+
+class MedicationAdministrationSpec(BaseMedicationAdministrationSpec):
+    @field_validator("encounter")
+    @classmethod
+    def validate_encounter_exists(cls, encounter):
+        if not PatientConsultation.objects.filter(external_id=encounter).exists():
+            err = "Encounter not found"
+            raise ValueError(err)
+        return encounter
+
+    @field_validator("request")
+    @classmethod
+    def validate_request(cls, request):
+        if not MedicationRequest.objects.filter(external_id=request).exists():
+            err = "Medication Request not found"
+            raise ValueError(err)
+        return request
+
+    def perform_extra_deserialization(self, is_update, obj):
+        if not is_update:
+            obj.encounter = PatientConsultation.objects.get(
+                external_id=self.encounter
+            )  # Needs more validation
+            obj.patient = obj.encounter.patient
+            obj.request = MedicationRequest.objects.get(external_id=self.request)
+
+
+class MedicationAdministrationReadSpec(BaseMedicationAdministrationSpec):
```

Comment:
List does not need created_by and updated_by, move those to retrieve spec

Reply:
We are using created_by in the list spec in the frontend,

<img width="672" alt="image" src="https://github.com/user-attachments/assets/a9ae86c4-bcd0-48c1-aeb9-4a472afda0a3" />


_Actually this is similar to other specs, most of the list specs have them._

Reply:
We will clean up all specs with created_by in them, the joins does not make sense. If updated by is not being used remove it.